From 0eb02e0b5cbd95e2727b427b75cc84229f3533e0 Mon Sep 17 00:00:00 2001
From: Shawn Rutledge <shawn.rutledge@qt.io>
Date: Tue, 23 Sep 2025 16:12:39 +0200
Subject: [PATCH 216/553] CRA review: plugins/platforminputcontexts/*

The files that we maintain by hand are marked as significant.
- D-Bus is security critical, but the Qt I-Bus implementation is merely
  a user, and does not do any parsing or communication on its own. The
  last known vulnerability CVE-2019-14822 was in the server, not in Qt.
- Likewise, QComposeInputContext uses xkb_compose functionality, so any
  risk is there, not in Qt.
- Both of these plugins work with keyboard events, but we do not
  consider that a risk in general.
- The remaining source files in the ibus directory are auto-generated,
  with a comment explaining that, so we do not modify them. If a
  vulnerability were found, we would need to fix qdbusxml2cpp first
  and then re-generate these.

QUIP: 23
Fixes: QTBUG-135725
Pick-to: 6.8
Change-Id: Ie4b0ef3d8151406ab5ddc758098d7871f320fe89
Reviewed-by: Liang Qi <liang.qi@qt.io>
(cherry picked from commit ad6afabdb244cf9bab9dc2767442c3fe17b1cf13)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../compose/qcomposeplatforminputcontext.cpp                    | 2 ++
 .../compose/qcomposeplatforminputcontext.h                      | 2 ++
 .../compose/qcomposeplatforminputcontextmain.cpp                | 1 +
 src/plugins/platforminputcontexts/ibus/main.cpp                 | 1 +
 .../platforminputcontexts/ibus/qibusplatforminputcontext.cpp    | 2 ++
 .../platforminputcontexts/ibus/qibusplatforminputcontext.h      | 2 ++
 src/plugins/platforminputcontexts/ibus/qibustypes.cpp           | 1 +
 src/plugins/platforminputcontexts/ibus/qibustypes.h             | 2 ++
 8 files changed, 13 insertions(+)

diff --git a/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontext.cpp b/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontext.cpp
index 3e741890768..a32d3556831 100644
--- a/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontext.cpp
+++ b/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontext.cpp
@@ -1,5 +1,7 @@
 // Copyright (C) 2019 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
+
 #include "qcomposeplatforminputcontext.h"
 
 #include <QtCore/QCoreApplication>
diff --git a/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontext.h b/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontext.h
index 79b98de3fa8..de08516b65a 100644
--- a/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontext.h
+++ b/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontext.h
@@ -1,5 +1,7 @@
 // Copyright (C) 2019 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
+
 #ifndef QCOMPOSEPLATFORMINPUTCONTEXT_H
 #define QCOMPOSEPLATFORMINPUTCONTEXT_H
 
diff --git a/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontextmain.cpp b/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontextmain.cpp
index 804ced957b9..6fc8bc3897d 100644
--- a/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontextmain.cpp
+++ b/src/plugins/platforminputcontexts/compose/qcomposeplatforminputcontextmain.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2019 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <qpa/qplatforminputcontextplugin_p.h>
 
diff --git a/src/plugins/platforminputcontexts/ibus/main.cpp b/src/plugins/platforminputcontexts/ibus/main.cpp
index d74be4bedf6..0f4b3985fd3 100644
--- a/src/plugins/platforminputcontexts/ibus/main.cpp
+++ b/src/plugins/platforminputcontexts/ibus/main.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include <qpa/qplatforminputcontextplugin_p.h>
 #include <QtCore/QStringList>
diff --git a/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.cpp b/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.cpp
index d70c3ccfe2d..06f7151f6df 100644
--- a/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.cpp
+++ b/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.cpp
@@ -1,5 +1,7 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
+
 #include "qibusplatforminputcontext.h"
 
 #include <QDebug>
diff --git a/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.h b/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.h
index 2314fb81410..1892ef0568a 100644
--- a/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.h
+++ b/src/plugins/platforminputcontexts/ibus/qibusplatforminputcontext.h
@@ -1,5 +1,7 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
+
 #ifndef QIBUSPLATFORMINPUTCONTEXT_H
 #define QIBUSPLATFORMINPUTCONTEXT_H
 
diff --git a/src/plugins/platforminputcontexts/ibus/qibustypes.cpp b/src/plugins/platforminputcontexts/ibus/qibustypes.cpp
index 2057ac154d1..d6f783ef9b0 100644
--- a/src/plugins/platforminputcontexts/ibus/qibustypes.cpp
+++ b/src/plugins/platforminputcontexts/ibus/qibustypes.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qibustypes.h"
 
diff --git a/src/plugins/platforminputcontexts/ibus/qibustypes.h b/src/plugins/platforminputcontexts/ibus/qibustypes.h
index 7940a7a3586..b10b5edd3c8 100644
--- a/src/plugins/platforminputcontexts/ibus/qibustypes.h
+++ b/src/plugins/platforminputcontexts/ibus/qibustypes.h
@@ -1,5 +1,7 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
+
 #ifndef QIBUSTYPES_H
 #define QIBUSTYPES_H
 
-- 
2.51.2

